pkg/install/1-installresources_test.go

```diff
@@ -0,0 +1,63 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/types"
+	aztypes "github.com/openshift/installer/pkg/types/azure"
+)
+
+func TestZones(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		zones    []string
+		replicas *int64
+		wantErr  string
+	}{
+		{
+			name:     "One replica in one zone",
+			zones:    []string{"1"},
+			replicas: to.Int64Ptr(1),
+		},
+		{
+			name:     "Two replicas, one in each of two zones",
+			zones:    []string{"1", "2"},
+			replicas: to.Int64Ptr(2),
+		},
+		{
+			name:     "One replica in two zones",
+			zones:    []string{"1", "2"},
+			replicas: to.Int64Ptr(1),
+			wantErr:  "cluster creation with 2 zones is unimplemented",
+		},
+		{
+			name:     "Two replicas in one zone",
+			zones:    []string{"1"},
+			replicas: to.Int64Ptr(2),
+		},
```

Comment:
you can hard-code replicas = 3 in this test.  I think the test cases that make sense here are:
* 3 replicas, no zones (this is actually manifested as `zones: []string{""}` due to installer weirdness)
* 3 replicas, 3 zones
* 3 replicas, 2 zones -> error

Reply:
@jim-minter There is a special empty case for 1 zone - should that be tested as well?